Add claimbans

TauCu opened the pull request:
This PR adds Claim Bans to GriefPrevention

I've added claimbans to GP to prevent players from annoying others inside claims (such as by inventory bombing, filling hopper systems, stealing loot from automatic farms etc)

Added commands:
- `claimban` aliases `banish`
- `claimunban` aliases `claimdeban, unbanish`
- `claimbanlist` aliases `claimbans`

Added Messages:
- `BannedFromClaim`
- `CannotBanManager`
- `PlayerBannedFromClaim`
- `PlayerBannedFromClaims`
- `PlayerUnBannedFromClaims`
- `PlayerUnBannedFromClaim`
- `BanListNoClaim`
- `BanListHeader`

Added Tasks:
- `CheckClaimbannedTask` interval `10` - Used to prevent mount plugins from bypassing claimbans

Added Data:
- List of claimbanned Player IDs mysql: `bannedplayerids TEXT` yaml: `bannedPlayerIDs`

TauCu committed:
Add claimbans

TauCu committed:
Improve claimbans

TauCu committed:
Merge branch 'TechFortress:master' into master

TauCu committed:
Fix execution order of claimban saving

Ashwin-droid reviewed (dismissed):
Wow
would love this feature

TauCu committed:
Improve efficiency of claimban lookups by removing string conversion & comparison of UUIDs

This is now implemented via a HashSet that can take advantage of UUID's hashcode & equals impl

TauCu committed:
better claimban ejection logic for subdivisions

TauCu committed:
ensure that both eject methods don't void players in the end

TauCu committed:
don't send player to bed spawn location if they are banned from it

TauCu committed:
don't eject players when they are ignoring claims when someone claimbans them

TauCu committed:
Fix DataBaseDataStore implementation for claimbans

TauCu committed:
Add unintrusive database migration for pre-claimbans databases
